jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/paths/PathCollection.java

```diff
@@ -0,0 +1,291 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.paths;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.StringTokenizer;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import org.eclipse.jetty.util.StringUtil;
+
+/**
+ * Generic Loader for Multiple Path references.
+ * <p>
+ * This fits in the same space as the ResourceCollection of past Jetty versions.
+ * </p>
+ */
+public class PathCollection extends ArrayList<Path> implements AutoCloseable
+{
+    /**
+     * Parse a delimited String of resource references and
+     * return a PathCollection it represents.
+     *
+     * <p>
+     * Supports glob references that end in {@code /*} or {@code \*}.
+     * Glob references will only iterate through the level specified and will not traverse
+     * found directories within the glob reference.
+     * </p>
+     *
+     * @param resources the comma {@code ,} or semicolon {@code ;} delimited
+     * String of path references.
+     * @param globDirs true to return directories in addition to files at the level of the glob
+     * @return the PathCollection parsed from input string.
+     */
+    public static PathCollection fromList(String resources, boolean globDirs) throws IOException
```

Comment:
What uses this?  I'm guessing our XML???
We should try doing it in XML without this helper method

Reply:
The old WebAppContext extraClasspath logic has this, it was moved into here, just to support the "*" logic that is present when dealing with classloader style of reference lists (vs resource style reference lists)

Reply:
This is the replacement of `Resource.fromList()`

See: https://github.com/eclipse/jetty.project/blob/jetty-11.0.9/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java#L946-L1018